### PR TITLE
feat(bigquery): add `unnest` and `pivot_longer` support

### PR DESCRIPTION
**Pull request by cpcloud**
This PR implements unnest support for the BigQuery backend.

Yep, it's a bit of a hack, and similar to how we're hacking unnest in for
backends like trino and snowflake, that only allow unnest in the `FROM` clause
of a `SELECT` statement.

Also included is a fix to handle invalid BigQuery names, which changes a bunch of the generated SQL.

What's really nice about implementing unnest for BigQuery is that `pivot_longer` now "just works".

### CODE REVIEW

**Comment by cpcloud** [comment]
Cloud backends look good:

```
cloud in 🌐 albatross in …/ibis on  bigquery-unnest is 📦 v5.0.0 via 🐍 v3.10.10 via ❄️  impure (ibis-3.10.10-env) took 6s
❯ pytest -m 'snowflake or bigquery' -n auto --dist=loadgroup --snapshot-update --allow-snapshot-deletion -q
bringing up nodes...
xx.........x..x........xx..x...x.x...x......x..............................................................x.. [  5%]
.........................................................................x.....x..x....x........x....xx.xx.... [ 10%]
..xx.........x...x.x.x.x....xxx..x.x..x....................xx....x.x...x.xx..x.x.xx..x..xx..x...xx.x.xxx..x..x [ 16%]
..x....xx..xx....x...x......xx.x...xx....x...xx....xx.....x.....x............x........xx..x....x.....xx......x [ 21%]
.....x.....x...x.x..........x.......................x......................................................... [ 27%]
.......x............................x........x..x.........x....x.x...x.x...xxx...xxx.x.xxxxxxxxxxxx.xxxxxxxxxx [ 32%]
xxxx.xxx..x..x....xx.xx.x.x...xx...xxxx..x.......x......xxx.....x.x....x....x.................x...x....x...x.. [ 38%]
x.xx..x..x.......xx.x.x.......xx....x...........x.x................x.........................xxxxxxxx.x....... [ 43%]
.........x..x......xx...xx............x.xx...........xx........x....x...............x.........x..........x.... [ 48%]
..x.x......................x..x......................xx.........x......x........x......s............xs...x.... [ 54%]
....x.......x...xx...x.xx.s....x..xx....x...x...xx.....xx...x.x.x.x......x...x.x...xx.x.x........xxxx.....x... [ 59%]
x..x...xx...x....................x.x........x..............x....x.....x.xx......x........x......x............. [ 65%]
.................x.........x............x...................................................x............x.... [ 70%]
.......x..................................s..........xx.............xxx......................xxxxxx.xxxxx.x.xx [ 76%]
...x....x........xxx.....x......x.....x.x....x......x..x.x....x....x...........................x.......xx...x. [ 81%]
...........x......xx..............x....x......................x.....xx...................x...............x...s [ 86%]
.......x...................................................................................................... [ 92%]
...............................x..............................................................s............... [ 97%]
x..xx.x..xx................................                                                                    [100%]
1694 passed, 6 skipped, 323 xfailed in 199.25s (0:03:19)
```

File: ibis/expr/operations/relations.py [hunk 1]
```diff
@@ -585,6 +585,15 @@ def schema(self):
         return backend._get_schema_using_query(self.query)
 
 
+class _UnnestTable(TableNode):
```
Comment (kszucs) [inline ibis/expr/operations/relations.py:588]: Can you add a comment about where is it used and why we need it?

Comment (cpcloud) [reply ibis/expr/operations/relations.py:588]: Yep!

Comment (cpcloud) [reply ibis/expr/operations/relations.py:588]: Done

**Comment by cpcloud** [comment]
There are some issues with chaining expressions that I still need to fix.

**Comment by cpcloud** [comment]
Closing this out for now. Will revisit for the next release.

**Pull request closed without merging**